doc/specs/#642 - Buffer Exporting and Logging/#642 - Buffer Exporting and Logging.md

```diff
@@ -0,0 +1,380 @@
+
+_figure 3: ConEmu settings_
+
+These applications all offer some settings in common. Primarily, the important
+feature is the ability to specify a path to a log file which contains some
+special string formatting. This allows the user to log to different files based
+on the time & date of the session, or based on the session name.
+
+### User Stories
+
+* **Story A**: The user is able to use a context menu entry on the tab to export
+  the contents of the buffer to a file, which they are prompted for.
+  - This is explicitly what was requested in [#642]
+* **Story B**: The user can bind an action to export the contents of the buffer
+  to a file, which they are prompted for.
+  - Very similar to **A**, but via the command palette or a keybinding.
+* **Story C**: The user can export to an explicit file via an action
+  - similar to **B**, but allowing for declaring the path to a file rather than
+    prompting at runtime.
+* **Story D**: The user can choose to append to a file when exporting, rather
+  than overwritting.
+* **Story E**: The user can specify a format string in the path to the file to
+  export to, which the Terminal will automatically replace with variables like
+  the time, date, and profile name.
+* **Story F**: When opening a specific profile, the user can automatically log
+  to a file
+* **Story G**: The user can execute an action to start or stop logging to a
+  given file.
+
+
+## Solution Design
+
+I'm proposing the following actions and profile settings
+
+* New Action: `exportBuffer()`.
+  - Export the contents of the buffer to a file.
+  - `path` (string, defaults to `""`): When empty, prompt the user for a name of
+    a file to export to, using a file picker. This path accepts special
+    formatting strings that will be substituted with certain variables
+    (discussed [below](#path-formatting)).
+  - `append` (boolean, defaults to `false`): When `false`, the file's contents
+    will be overwritten. When `true`, the buffer contents will be appended to
+    the end of the file.
+* New Profile Settings object: `logSettings`
+  - This is an object that describes a set of behavior for logging a profile.
+  - `path`: Same as the `path` in the `ExportBufferArgs` above
+  - `append`: Same as the `append` in the `ExportBufferArgs` above
+  - `captureAllOutput`: (boolean, defaults to `false`) When true, don't log only
+    printable characters, also log non-printable escape characters written to
+    the Terminal.
+  - `captureInput`: (boolean, defaults to `false`) Additionally log input to the
+    Terminal to the file. Input will be formatted as the traditional VT
+    sequences, rather than the full `win32-input` encoding.
+  - `newFileEveryDay`: (boolean, defaults to `false`) This requires the `day` to
+    be an element of the path format string. When logging with this setting,
+    opens a new file at midnight and starts writing that one.
+  <!-- TODO! - `flushFrequently`: (boolean, defaults to `true`) -->

```

Comment:
Meh, I found it in the PuTTY docs, thought it was interesting, but commented out because I hadn't investigated it, then forgot to remove the comments before shipping the PR. Will move to future considerations, since it doesn't seem _critical_ to the design.